qiskit_nature/transformers/second_quantization/electronic/seniority_zero_transformer.py

```diff
@@ -0,0 +1,242 @@
+
+            transformed_property.second_q_ops = partial(  # type: ignore[assignment]
+                second_q_ops_electronic_energy, transformed_property
+            )
+
+        elif isinstance(prop, ParticleNumber):
+            transformed_property = prop
+
+            transformed_property.second_q_ops = partial(  # type: ignore[assignment]
+                second_q_ops_particle_number, transformed_property
+            )
+        elif isinstance(prop, DriverMetadata):
+            transformed_property = prop
+        elif isinstance(prop, SecondQuantizedProperty):
+            transformed_property = None
+        elif isinstance(prop, ElectronicBasisTransform):
+            transformed_property = None
+
+        else:
+            raise TypeError(f"{type(prop)} is an unsupported Property-type for this Transformer!")
+
+        return transformed_property
+
+    def _restricted_electronic_integrals(
+        self, h_1: np.ndarray, h_2: np.ndarray
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Transform PyScf integrals to the restricted formalism
+        arxiv: 2002.00035, Appendix A, Equations (A10),(A11),(A12)
+        https://arxiv.org/pdf/2002.00035.pdf
+
+        Args:
+            h_1 (np.ndarray): one-body integrals, (N x N)
+            h_2 (np.ndarray): two-body integrals, (N x N x N x N)
+
+        Returns:
+            hr_1 (np.ndarray): restricted one-body integrals, (N x N)
+            hr_2 (np.ndarray): restricted two-body integrals, (N x N)
+        """
+        # change of notation (may only be valid for PySCF -> restricted formalism)
+        h_2 = np.einsum("ijkl->ljik", h_2)
+
+        hr_1 = np.zeros((self._num_modes, self._num_modes))
+        hr_2 = np.zeros((self._num_modes, self._num_modes))
+
+        for i in range(self._num_modes):
+            hr_1[i, i] = 2 * h_1[i, i] + h_2[i, i, i, i]
+
+            for j in range(self._num_modes):
+                if i != j:
+                    hr_1[i, j] = h_2[i, i, j, j]
+                    hr_2[i, j] = 2 * h_2[i, j, j, i] - h_2[i, j, i, j]
+
+        return hr_1, hr_2
+
+
+def second_q_ops_electronic_energy(self: ElectronicEnergy) -> ListOrDictType[VibrationalOp]:
+    """Creates the second quantization operators in the restricted formalism
+    arxiv: 2002.00035, Equation (2)
+    https://arxiv.org/pdf/2002.00035.pdf
+    """
+
+    hr_1 = self.get_electronic_integral(ElectronicBasis.MO, 1).get_matrix()
+    hr_2 = self.get_electronic_integral(ElectronicBasis.MO, 2).get_matrix()
+
+    num_modes = hr_1.shape[0]
+
+    label_m = lambda x: "I" * x + "-" + "I" * (num_modes - x - 1)
+    label_p = lambda x: "I" * x + "+" + "I" * (num_modes - x - 1)
+
+    # Create pair creation and annihilation operators for the spatial orbitals
+    b = [VibrationalOp(label_m(i), num_modes=num_modes, num_modals=1) for i in range(num_modes)]
+    b_dag = [VibrationalOp(label_p(i), num_modes=num_modes, num_modals=1) for i in range(num_modes)]
+
+    # Build the operators
```

Comment:
```suggestion
    # TODO: refactor to construct `VibrationalOp` from a list of labels rather than summing multiple instances
    # Build the operators
```
This can be improved significantly in terms of performance in a similar style to #638. However, the `VibrationalOp` is no longer on par with the `FermionicOp` so I am not sure if this is already possible now. Please leave this TODO comment in the code for the time being.

Note: you will need to reformat the file to avoid the line length limit being exceeded.